module/VuFind/src/VuFind/ILS/Connection.php

```diff
@@ -1227,4 +1289,87 @@ public function __call($methodName, $params)
             'Cannot call method: ' . $this->getDriverClass() . '::' . $methodName
         );
     }
+
+    /**
+     * Get data for an ILS method from shared or session cache
+     *
+     * @param string $methodName The name of the called method.
+     * @param array  $params     Array of passed parameters.
+     *
+     * @return ?array
+     */
+    protected function getCachedData($methodName, $params)
+    {
+        $cacheLifeTime = $this->cacheLifeTime[$methodName] ?? null;
+        $cacheStorage = $this->cacheStorage[$methodName] ?? null;
+        if (!$cacheLifeTime || !$cacheStorage) {
+            return null;
+        }
+        $cacheKey = $methodName . md5(serialize($params));
```

Comment:
Would it be better to pass in the $cacheKey instead of the $params both here and in the put method? This way, you would only need to calculate the hash in one place, which would both improve efficiency (no double hashing/serialization needed) and reduce redundancy (no chance of the algorithm somehow getting out of sync due to code changes). It would also be possible to create a separate support method for generating the key, in case that would be a useful customization point.

Reply:
Or perhaps even better, the getCacheKey method could also contain the logic to determine whether caching is supported and return null if it is not. Then more logic could be centralized/deduplicated.

Reply:
I refactored it a bit further. What do you think?